File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        int longest;
        unordered_set<int> set;
        for(int i = 0 ; i < nums.size() ; i++){
            set.insert(nums[i]);
        }

        for(int it : set){
            if(set.find(it-1) == set.end()){
                int c = 1;
                int x = it+1;
                while(set.find(x) != set.end()){
                    c++;
                    x++;
                }
                longest = max(longest, c);
            }
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0;
        Arrays.sort(nums);
        int c = 0, mc =0 ;
        for(int i = 0 ; i < nums.length-1 ; i++){
            if(nums[i+1] == nums[i]) continue;
            if(nums[i+1] - nums[i] == 1 ){
                c++;
                if(c > mc)
                    mc = c;
            }
            else
                c =0;
        }
        return mc+1;
    }
}